### PR TITLE
Document auto-loading custom test execution listeners.

### PR DESCRIPTION
**Pull request by sormuras**
## Overview

Initial documentation for #583 done in User Guide and Release Notes.

---

I hereby agree to the terms of the JUnit Contributor License Agreement.

### CODE REVIEW

File: documentation/src/docs/asciidoc/launcher-api.adoc [hunk 1]
```diff
@@ -80,3 +80,13 @@ is currently supported via Java's `java.util.ServiceLoader` mechanism. For examp
 `junit-jupiter-engine` module registers its `org.junit.jupiter.engine.JupiterTestEngine`
 in a file named `org.junit.platform.engine.TestEngine` within the `/META-INF/services` in
 the `junit-jupiter-engine` JAR.
+
+[[launcher-api-listeners-custom]]
+==== Plugging in Your Own Test Execution Listeners
+
+In addition to the public {Launcher} API method for registering test execution listeners
+programmatically, custom {TestExecutionListener} implementations discovered at runtime via
+Java's ServiceLoader facility are automatically registered with the {Launcher}. For example,
```
Comment (marcphilipp) [inline documentation/src/docs/asciidoc/launcher-api.adoc]: This should reference `DefaultLauncher`, right?

Comment (sormuras) [inline documentation/src/docs/asciidoc/launcher-api.adoc]: Ja. The first is {Launcher}, the second {DefaultLauncher}.

File: documentation/src/docs/asciidoc/launcher-api.adoc [hunk 2]
```diff
@@ -80,3 +80,13 @@ is currently supported via Java's `java.util.ServiceLoader` mechanism. For examp
 `junit-jupiter-engine` module registers its `org.junit.jupiter.engine.JupiterTestEngine`
 in a file named `org.junit.platform.engine.TestEngine` within the `/META-INF/services` in
 the `junit-jupiter-engine` JAR.
+
+[[launcher-api-listeners-custom]]
+==== Plugging in Your Own Test Execution Listeners
+
+In addition to the public {Launcher} API method for registering test execution listeners
+programmatically, custom {TestExecutionListener} implementations discovered at runtime via
+Java's ServiceLoader facility are automatically registered with the {DefaultLauncher}. For
```
Comment (sbrannen) [inline documentation/src/docs/asciidoc/launcher-api.adoc]: ServiceLoader should be formatted.

File: documentation/src/docs/asciidoc/launcher-api.adoc [hunk 3]
```diff
@@ -80,3 +80,13 @@ is currently supported via Java's `java.util.ServiceLoader` mechanism. For examp
 `junit-jupiter-engine` module registers its `org.junit.jupiter.engine.JupiterTestEngine`
 in a file named `org.junit.platform.engine.TestEngine` within the `/META-INF/services` in
 the `junit-jupiter-engine` JAR.
+
+[[launcher-api-listeners-custom]]
+==== Plugging in Your Own Test Execution Listeners
+
+In addition to the public {Launcher} API method for registering test execution listeners
+programmatically, custom {TestExecutionListener} implementations discovered at runtime via
```
Comment (sbrannen) [inline documentation/src/docs/asciidoc/launcher-api.adoc]: Please note that we consistently format Java types using a monospace font (e.g., `{TestExecutionListener}`). The curly brackets denote that it is a link that we have defined globally for the documentation. If a link has not been defined for the type, consider creating one or omitting the curly brackets, but do not omit the  back ticks.

Please apply this throughout your Asciidoc changes.

File: documentation/src/docs/asciidoc/launcher-api.adoc [hunk 4]
```diff
@@ -80,3 +80,13 @@ is currently supported via Java's `java.util.ServiceLoader` mechanism. For examp
...
+Java's ServiceLoader facility are automatically registered with the {DefaultLauncher}. For
+example, a `example.TestInfoPrinter` class implementing `TestExecutionListener` and
```
Comment (sbrannen) [inline documentation/src/docs/asciidoc/launcher-api.adoc]: "a" --> "an"

File: documentation/src/docs/asciidoc/launcher-api.adoc [hunk 5]
```diff
@@ -80,3 +80,13 @@ is currently supported via Java's `java.util.ServiceLoader` mechanism. For examp
...
+enlisted within the `/META-INF/services/org.junit.platform.launcher.TestExecutionListener`
```
Comment (sbrannen) [inline documentation/src/docs/asciidoc/launcher-api.adoc]: "enlisted" --> "declared"

File: documentation/src/docs/asciidoc/release-notes-5.0.0-M4.adoc [hunk 6]
```diff
@@ -26,7 +26,8 @@ on GitHub.
 
 ===== New Features
 
-* ❓
+* Support for loading and registering custom TestExecutionListener
```
Comment (sbrannen) [inline documentation/src/docs/asciidoc/release-notes-5.0.0-M4.adoc]: back ticks, please. 😉

**Comment by sbrannen** [comment]
This has been merged into `master`.

Thanks for applying the requested changes.

**Pull request merged**